### PR TITLE
[IMPAC-578] Create Widget KPIs

### PR DESCRIPTION
**Pull request by xaun**
@cesar-tonnoir please review. Marked WIP as the #update & #delete actions are not validated as still working / updated.

### CODE REVIEW

**Comment by ouranos** [comment]
Also, are we missing specs here?

**Comment by xaun** [comment]
Linked PRs:
maestrano/maestrano-hub#514
https://github.com/maestrano/impac-angular/pull/338

**Comment by xaun** [comment]
@ouranos can you please re-review. The "backward compatiblity fix" for mnoe is a bit strange, when mno-hub is without the changes is maestrano-hub#515, mnoe kpi template errors with `ActionView::Template::Error (undefined method `map' for #<MnoEnterprise::Impac::Alert(alerts) >`. I would expected the her `:has_many` relationship to manage this? Let me know if there is a better way to handle / guard this. 
Backwards compatibility with impac-angular confirmed.

File: api/spec/controllers/mno_enterprise/jpi/v1/impac/kpis_controller_spec.rb [hunk 1]
```diff
@@ -75,46 +75,77 @@ module MnoEnterprise
...
+            api_stub_for(get: "/users/#{user.id}/alerts", response: from_api({}))
+          end
 
-      it "creates the kpi" do
-        subject
-        expect(assigns(:kpi)).to eq(kpi)
+          it "creates kpi alerts" do
+            subject
+            expect(assigns(:kpi).alerts).to eq([alert])
+            expect(response.code).to eq('200')
```
Comment (ouranos) [inline api/spec/controllers/mno_enterprise/jpi/v1/impac/kpis_controller_spec.rb]: `expect(response).to have_http_status(:ok)`

File: api/spec/controllers/mno_enterprise/jpi/v1/impac/kpis_controller_spec.rb [hunk 2]
```diff
@@ -75,46 +75,77 @@ module MnoEnterprise
...
+          end
+        end
+
+        it { subject; expect(response.code).to eq('200') }
```
Comment (ouranos) [inline api/spec/controllers/mno_enterprise/jpi/v1/impac/kpis_controller_spec.rb]: `it { is_expected.to have_http_status(:ok) }`

**Pull request merged**